arwen/elrondapi/bigIntOps.go

```diff
@@ -399,14 +399,20 @@ func v1_4_bigIntStorageLoadUnsigned(context unsafe.Pointer, keyOffset int32, key
 	runtime := arwen.GetRuntimeContext(context)
 	storage := arwen.GetStorageContext(context)
 	metering := arwen.GetMeteringContext(context)
+	enableEpochsHandler := arwen.GetVMHost(context).EnableEpochsHandler()
 
 	key, err := runtime.MemLoad(keyOffset, keyLength)
 	if arwen.WithFault(err, context, runtime.BigIntAPIErrorShouldFailExecution()) {
 		return -1
 	}
 
-	bytes, usedCache := storage.GetStorage(key)
-	storage.UseGasForStorageLoad(bigIntStorageLoadUnsignedName, metering.GasSchedule().BigIntAPICost.BigIntStorageLoadUnsigned, usedCache)
+	bytes, trieDepth, usedCache := storage.GetStorage(key)
+	blockchainLoadCost, err := arwen.GetStorageLoadCost(int64(trieDepth), metering, metering.GasSchedule().BigIntAPICost.BigIntStorageLoadUnsigned, enableEpochsHandler)
```

Comment:
if moved to storage, maybe you can do the computation only if usedCache was false on UseGasForStorageLoad? 
(this will mean that you add the trie depth on UseGasForStorageLoad - not sure if that's easy to change)

Reply:
In the func `v1_4_getCurrentESDTNFTNonce` `UseGasAndAddTracedGas` is used to consume gas instead of `UseGasForStorageLoad`, which does not take into consideration if the `usedCache` is `true`